### PR TITLE
Update Dependencies

### LINKED ISSUE #107: Compatibility with python-socketio 5.1.0
The python socket io lib has integrated the socketio_client lib. 
https://python-socketio.readthedocs.io/en/latest/client.html

But using the latest version (5.1.0) like in "Client Examples" resolves into an error "Unexpected server response" from slurk. https://python-socketio.readthedocs.io/en/latest/intro.html 

```
import socketio

sio = socketio.Client()

@sio.event
def connect():
    print('connection established')

@sio.event
def my_message(data):
    print('message received with ', data)
    sio.emit('my response', {'response': 'my response'})

@sio.event
def disconnect():
    print('disconnected from server')

sio.connect('http://localhost:5000')
sio.wait()
```

The problem might be that the message returned by slurk is somehow zero bytes long (thats visible in the logs). Usually the message is at least two bytes long (using socketio_client 0.7.2).

### PR DESCRIPTION
**Pull request by TimDiekmann**
- Update SocketIO to 5.0
- Update SocketIO Javascript to 4.0
- Update JQuery to 3.6
- Update SQLAlchemy to 1.4
- Only freeze used dependencies

This is a list of compatible versions of SocketIO:
https://flask-socketio.readthedocs.io/en/latest/#version-compatibility

closes #107

### CODE REVIEW

**Comment by phisad** [comment]
Looks good. In addition, (if I see correctly) a connecting user is now receiving a `joined_room` event for all rooms that were joined.

File: app/login/events.py [hunk 1]
```diff
@@ -33,8 +33,11 @@ def connect():
         log_event("join", current_user, room)
     db.session.commit()
 
+    for room in current_user.current_rooms:
+        socketio.emit("joined_room", dict(user=current_user.id, room=room.name), room=request.sid)
 
-@socketio.on('ready')
+
+# @socketio.on('ready')
```
Comment (janagoetze) [inline app/login/events.py:40]: do you still need this?

**Review by janagoetze — APPROVED** [review]
only non-blocking comment

File: app/login/events.py (hunk 1 above)
Comment (TimDiekmann) [reply app/login/events.py:40]: Weirdly I did not get a notification for your comment. 
I like to keep it for now until we are sure we really don't need this anymore.

**Pull request closed without merging**